### PR TITLE
osdocs-617 FIPS

### PR DESCRIPTION
**Pull request by kalexand-rh**
Draft for https://jira.coreos.com/browse/OSDOCS-617

Draft build here: http://file.rdu.redhat.com/kalexand/120519/osdocs617/installing/installing-fips.html

### CODE REVIEW

File: installing/installing-fips.adoc [hunk 1]
```diff
@@ -0,0 +1,74 @@
+[id="installing-fips"]
+= FIPS compliant {product-title} clusters
+include::modules/common-attributes.adoc[]
+:context: installing-fips
+
+toc::[]
+
+Starting with version 4.3, you can install an {product-title} cluster that runs in a FIPS compliant mode.
```
Comment (mrunalp) [inline installing/installing-fips.adoc]: clusters that use FIPS validated crypto libraries

File: installing/installing-fips.adoc [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+[id="installation-about-fips-components_{context}"]
+==  FIPS compliance in components that the cluster uses
+
+Although the {product-title} cluster itself uses FIPS validated modules, ensure that the systems that support your {product-name} cluster use FIPS validated modules for cryptography.
+
+[id="installation-about-fips-components-etcd_{context}"]
+=== etcd
+
+To ensure that the secrets that are stored in etcd use FIPS validated encryption, encrypt the etcd volume by using RHEL-provided disk encryption.
```
Comment (xiaojiey) [inline installing/installing-fips.adoc]: What do you really want to talk about? The disk encryption or encryption of secrets stored in etcd(in story https://jira.coreos.com/browse/MSTR-485)?If you are talking about disk encryption, will you add the encryption of secrets stored in etcd? Thanks. Besides, OCP QE didn't do special test to cover disk encryption.

**Comment by pdhamdhe** [comment]
@kalexand-rh Open this issue https://github.com/openshift/openshift-docs/issues/18430 to correct RHOCS version in "FIPS validation in OpenShift Container Platform" section.

File: installing/installing-fips.adoc (hunk 2 above)
Comment (kalexand-rh) [reply installing/installing-fips.adoc]: @knewcomerRH, which angle do you want to cover?

**Comment by kalexand-rh** [comment]
@pdhamdhe, thanks! If you have more feedback, please leave a comment on the PR instead of opening an issue. I'm going to address all the feedback before I merge the content.

**Comment by pdhamdhe** [comment]
@kalexand-rh Sure. Thanks!

File: installing/installing-fips.adoc [hunk 3]
```diff
@@ -0,0 +1,74 @@
+[id="installing-fips"]
+= FIPS compliant {product-title} clusters
+include::modules/common-attributes.adoc[]
+:context: installing-fips
+
+toc::[]
+
+Starting with version 4.3, you can install an {product-title} cluster that runs in a FIPS compliant mode.
+
+For the {op-system-first} machines in your cluster, this change is applied when the machines are deployed based on the status of an option in the `install-config.yaml` file, which governs the cluster options that a user can change during cluster deployment. For Red Hat Enterprise Linux machines, you must configure the FIPS option when you install the operating system on the machines that you plan to use as worker machines. These configuration methods ensure that your cluster meet the requirements of a FIPS compliance audit: only FIPS validated cryptography packages are enabled before the initial system boot.
```
[the remark this answers is not in the document]
Comment (xiaojiey) [reply installing/installing-fips.adoc]: For "Red Hat Enterprise Linux machines, you must configure the FIPS option when you install the operating system on the machines that you plan to use as worker machines. ",  is it better to change it to "Red Hat Enterprise Linux machines, you must enable  FIPS mode when you install the operating system on the machines that you plan to use as worker machines"? Thanks. Besides, It's better to attach the link "https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/security_guide/chap-federal_standards_and_regulations#sec-Enabling-FIPS-Mode" to tell customer how to enable fips on rhel7.

File: installing/installing-fips.adoc (hunk 2 above)
Comment (knewcomerRH) [reply installing/installing-fips.adoc]: @kalexand-rh I would not mention the ability to encrypt the etcd datastore or to do NBDE in the FIPS section of the documentation. Just focus on FIPS. @simo5 what can we say about FIPS and the aes cbc encryption algorithm used to encrypt the etcd datastore?

Comment (simo5) [reply installing/installing-fips.adoc]: AES CBC is an approved algorithm so it should be fine ?
Not sure I understand the context thugh.

File: installing/installing-fips.adoc (hunk 3 above)
[the remark this answers is not in the document]
Comment (kalexand-rh) [reply installing/installing-fips.adoc]: I updated the language here. I think that if we want the link to the RHEL docs about FIPS in this topic that it makes more sense around line 74, so I added it there. I also have that link in the instructions about configuring the RHEL nodes.

File: installing/installing-fips.adoc (hunk 2 above)
Comment (kalexand-rh) [reply installing/installing-fips.adoc]: For now, I'm adding a link to the process for encrypting the etcd datastore: http://file.rdu.redhat.com/kalexand/120519/osdocs617/authentication/encrypting-etcd.html#encrypting-etcd

**Comment by kalexand-rh** [comment]
@xiaojiey, @knewcomerRH, @simo5, are there any other changes that you'd like to see?

File: installing/installing-fips.adoc (hunk 2 above)
Comment (knewcomerRH) [reply installing/installing-fips.adoc]: @kalexand-rh We should focus on encrypting the etcd datastore, not the RHCOS volumes here. And the aes cbc crypto algorithm which we use is FIPS approved.

Comment (kalexand-rh) [reply installing/installing-fips.adoc]: Thank you @knewcomerRH! I've updated this section to discuss the etcd datastore instead. Please let me know if you see anything else that needs to change.

**Comment by xiaojiey** [comment]
> @xiaojiey, @knewcomerRH, @simo5, are there any other changes that you'd like to see?

It looks good to me. Thanks a lot for your update.

**Comment by kalexand-rh** [comment]
@openshift/team-documentation PTAL?

File: installing/installing-fips.adoc [hunk 4]
```diff
@@ -0,0 +1,76 @@
+[id="installing-fips"]
+= FIPS compliant {product-title} clusters
+include::modules/common-attributes.adoc[]
+:context: installing-fips
+
+toc::[]
+
+Starting with version 4.3, you can install an {product-title} cluster that that use FIPS validated cryptographic libraries.
+
+For the {op-system-first} machines in your cluster, this change is applied when the machines are deployed based on the status of an option in the `install-config.yaml` file, which governs the cluster options that a user can change during cluster deployment. Red Hat Enterprise Linux machines, you must enable FIPS mode when you install the operating system on the machines that you plan to use as worker machines. These configuration methods ensure that your cluster meet the requirements of a FIPS compliance audit: only FIPS validated cryptography packages are enabled before the initial system boot.
```
Comment (ahardin-rh) [inline installing/installing-fips.adoc]: With Red Hat Enterprise Linux machines?

File: installing/installing-fips.adoc [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+[id="installation-about-fips-components-etcd_{context}"]
+=== etcd
+
+To ensure that the secrets that are stored in etcd use FIPS validated encryption, encrypt the etcd datastore by using a FIPS-approved cryptographic algorithm. After you install the cluster, you can xref:../authentication/encrypting-etcd.adoc#encrypting-etcd[encrypt the etcd data] by using the `aes cbc` algorithm.
+
+[id="installation-about-fips-components-storage_{context}"]
+=== Storage
+
+For local storage, use RHEL-provided disk encryption or Container Native Storage that uses RHEL-provided disk encryption. By storing all data in volumes that use RHEL-provided disk encryption and enabling FIPS mode for your cluster, both data at rest and data in motion, or network data, are  protected by FIPS validated encryption.
```
Comment (ahardin-rh) [inline installing/installing-fips.adoc]: possible extra space before protected

File: modules/installation-gcp-config-yaml.adoc [hunk 6]
```diff
@@ -77,7 +78,8 @@ sshKey: ssh-ed25519 AAAA... <6>
 If you disable simultaneous multithreading, ensure that your capacity planning accounts for the dramatically decreased machine performance. Use larger machine types, such as `n1-standard-8`, for your machines if you disable simultaneous multithreading.
 ====
 <5> To configure faster storage for etcd, especially for larger clusters, set the storage type as `io1` and set `iops` to `2000`.
-<6> You can optionally provide the `sshKey` value that you use to access the machines in your cluster.
+<6> Whether to enable or disable FIPS mode. By default, FIPS mode is not enabled. If FIPS mode is enabled, the {op-system-first} machines that {product-title} runs on bypass the default Kubernetes cryptography suite and use the FIPS validated cryptography modules that are provided with {op-system} instead.
```
Comment (ahardin-rh) [inline modules/installation-gcp-config-yaml.adoc]: s/Whether to enable or disable/Set to enable or disable

**Review by ahardin-rh — COMMENTED** [review]
Just a few minor comments from me. Otherwise, looks good!

**Pull request merged**